### PR TITLE
Handle race condition in the monitoring of the migration process on KVM

### PR DESCRIPTION
**Pull request by remibergsma**
This is issue is reported under these conditions:
- VM migration has physically finished
- Migration thread executor is still not marked as terminated

```
There is a race condition in the monitoring of the migration process on KVM. 
If the monitor wakes up in the tight window after the migration succeeds,
but before the migration thread terminates, the monitor will get a LibvirtException 
“Domain not found: no domain with matching uuid” when checking on the migration status. 
This in turn causes CloudStack to sync the VM state to stop, in which it issues a defensive 
StopCommand to ensure it is correctly synced.

Fix: Prevent LibvirtException: "Domain not found" caused by the call to dm.getInfo()
```

```
From the migration monitoring process, as VM has been migrated, that domain cannot 
be found (we should get the destination domain from the migration thread later) and the
LibvirtException is thrown: LibvirtException “Domain not found: no domain with matching uuid”.
So basically the idea is just logging the issue but not suspending the VM as the migration has
already been performed, but the migration thread is not marked as finished. Once the migrarion
thread is marked as finished, the migration monitoring process (while loop) ends and the migration
command wrapper would continue its execution.
Regarding your last question, we won't need to suspend the VM in this case as the migration has
already been performed. VM suspension will take place whenever the migration is in progress and
those conditions are met.
```
Thanks @nvazquez !

### CODE REVIEW

**Comment by sspans** [comment]
This seems to match management-server logging:

```
2018-02-23 09:56:51.566 WARN  [c.c.a.c.a.v.MigrateVMCmd] (logid: 9d073389) (ctx: 67ef55c2) (job: 4192168) Exception:
com.cloud.exception.VirtualMachineMigrationException: Caught exception while handling a VmWorkJob
        at com.cloud.vm.VirtualMachineManagerImpl.migrate(VirtualMachineManagerImpl.java:3705)
        at com.cloud.vm.UserVmManagerImpl.migrateVirtualMachine(UserVmManagerImpl.java:2968)
        at sun.reflect.GeneratedMethodAccessor518.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:333)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
        at com.cloud.event.ActionEventInterceptor.invoke(ActionEventInterceptor.java:34)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:168)
        at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:213)
        at com.sun.proxy.$Proxy210.migrateVirtualMachine(Unknown Source)
        at com.cloud.api.command.admin.vm.MigrateVMCmd.execute(MigrateVMCmd.java:129)
        at com.cloud.api.ApiDispatcher.dispatch(ApiDispatcher.java:124)
        at com.cloud.api.ApiAsyncJobDispatcher.runJob(ApiAsyncJobDispatcher.java:88)
        at com.cloud.framework.jobs.impl.AsyncJobManagerImpl$1.runInContext(AsyncJobManagerImpl.java:259)
        at com.cloud.managed.context.ManagedContextRunnable$1.run(ManagedContextRunnable.java:39)
        at com.cloud.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:38)
        at com.cloud.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:84)
        at com.cloud.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:35)
        at com.cloud.managed.context.ManagedContextRunnable.run(ManagedContextRunnable.java:36)
        at com.cloud.framework.jobs.impl.AsyncJobManagerImpl$1.run(AsyncJobManagerImpl.java:205)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: com.cloud.exception.CloudException: Caught exception while handling a VmWorkJob
        at com.cloud.vm.VmWorkJobHandlerProxy.handleVmWorkJob(VmWorkJobHandlerProxy.java:104)
        at com.cloud.vm.VirtualMachineManagerImpl.handleVmWorkJob(VirtualMachineManagerImpl.java:3422)
        at com.cloud.vm.VmWorkJobDispatcher.runJob(VmWorkJobDispatcher.java:89)
        at com.cloud.framework.jobs.impl.AsyncJobManagerImpl$1.runInContext(AsyncJobManagerImpl.java:259)
        at com.cloud.managed.context.ManagedContextRunnable$1.run(ManagedContextRunnable.java:39)
        at com.cloud.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:38)
        at com.cloud.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:84)
        at com.cloud.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:35)
        at com.cloud.managed.context.ManagedContextRunnable.run(ManagedContextRunnable.java:36)
        at com.cloud.framework.jobs.impl.AsyncJobManagerImpl$1.run(AsyncJobManagerImpl.java:205)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: com.cloud.utils.exception.CloudRuntimeException: Domain not found: no domain with matching uuid 'da0b65a5-9649-44d8-ab6d-e04857605eca' (i-57-1699-VM)
        at com.cloud.vm.VirtualMachineManagerImpl.migrate(VirtualMachineManagerImpl.java:2206)
        at com.cloud.vm.VirtualMachineManagerImpl.orchestrateMigrate(VirtualMachineManagerImpl.java:2105)
        at com.cloud.vm.VirtualMachineManagerImpl.orchestrateMigrate(VirtualMachineManagerImpl.java:2081)
        at sun.reflect.GeneratedMethodAccessor698.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at com.cloud.vm.VmWorkJobHandlerProxy.handleVmWorkJob(VmWorkJobHandlerProxy.java:88)
        ... 14 common frames omitted
2018-02-23 09:56:51.566 DEBUG [c.c.f.j.i.AsyncJobManagerImpl] (logid: 9d073389) (job: 4192168) Complete async job-4192168, jobStatus: FAILED, resultCode: 530, result: com.cloud.api.response.ExceptionResponse/null/{"uuidList":[],"errorcode":530,"errortext":"Caught exception while handling a VmWorkJob"}
```
